DiskIOMeter.c

```diff
@@ -138,30 +202,121 @@ static void DiskIOMeter_display(ATTR_UNUSED const Object* cast, RichString* out)
    int color = cached_utilisation_diff > 40.0 ? METER_VALUE_NOTICE : METER_VALUE;
    int len = xSnprintf(buffer, sizeof(buffer), "%.1f%%", cached_utilisation_diff);
    RichString_appendnAscii(out, CRT_colors[color], buffer, len);
+}
 
-   RichString_appendAscii(out, CRT_colors[METER_TEXT], " read: ");
-   RichString_appendAscii(out, CRT_colors[METER_VALUE_IOREAD], cached_read_diff_str);
-   RichString_appendAscii(out, CRT_colors[METER_VALUE_IOREAD], "iB/s");
+static void DiskIOMeter_updateValues(Meter* this) {
+   DiskIOMeterData* data = this->meterData;
 
-   RichString_appendAscii(out, CRT_colors[METER_TEXT], " write: ");
-   RichString_appendAscii(out, CRT_colors[METER_VALUE_IOWRITE], cached_write_diff_str);
-   RichString_appendAscii(out, CRT_colors[METER_VALUE_IOWRITE], "iB/s");
+   Meter_updateValues(data->diskIORateMeter);
+   Meter_updateValues(data->diskIOTimeMeter);
 }
 
-const MeterClass DiskIOMeter_class = {
+static void DiskIOMeter_draw(Meter* this, int x, int y, int w) {
+   DiskIOMeterData* data = this->meterData;
+
+   /* Use the same width for each sub meter to align with CPU meter */
+   const int colwidth = w / 2;
+   const int diff = w % 2;
+
+   assert(data->diskIORateMeter->draw);
+   data->diskIORateMeter->draw(data->diskIORateMeter, x, y, colwidth);
+   assert(data->diskIOTimeMeter->draw);
+   data->diskIOTimeMeter->draw(data->diskIOTimeMeter, x + colwidth + diff, y, colwidth);
```

Comment:
While initializing or stale, this looks kinda strange (in text mode).

The IO Time meter itself could benefit from actually labeling the value it shows in text mode.